src/Compiler/Service/FSharpProjectSnapshot.fs

```diff
@@ -72,6 +72,13 @@ type FSharpFileSnapshot(FileName: string, Version: string, GetSource: unit -> Ta
     static member Create(fileName: string, version: string, getSource: unit -> Task<ISourceTextNew>) =
         FSharpFileSnapshot(fileName, version, getSource)
 
+    static member CreateFromString(filename: string, content: string) =
+        FSharpFileSnapshot(
+            filename,
+            Md5Hasher.hashString content |> Md5Hasher.toString,
```

Comment:
Should probably eventually change to something more proper, can't ship with md5 hashing

Reply:
Most definitely. We should probably bring in some version of xxHash.

Reply:
Yeah. That was my thinking.